eclair-core/src/main/scala/fr/acinq/eclair/channel/Channel.scala

```diff
@@ -722,97 +722,97 @@ class Channel(val nodeParams: NodeParams, val wallet: OnChainChannelFunder, remo
    */
 
   when(NORMAL)(handleExceptions {
-    case Event(c: CMD_ADD_HTLC, d: DATA_NORMAL) if d.localShutdown.isDefined || d.remoteShutdown.isDefined =>
+    case Event(c: CMD_ADD_HTLC, DATA_NORMAL(d)) if d.localShutdown.isDefined || d.remoteShutdown.isDefined =>
       // note: spec would allow us to keep sending new htlcs after having received their shutdown (and not sent ours)
       // but we want to converge as fast as possible and they would probably not route them anyway
       val error = NoMoreHtlcsClosingInProgress(d.channelId)
       handleAddHtlcCommandError(c, error, Some(d.channelUpdate))
 
-    case Event(c: CMD_ADD_HTLC, d: DATA_NORMAL) =>
+    case Event(c: CMD_ADD_HTLC, DATA_NORMAL(d)) =>
       Commitments.sendAdd(d.commitments, c, nodeParams.currentBlockHeight, nodeParams.onChainFeeConf) match {
         case Right((commitments1, add)) =>
           if (c.commit) self ! CMD_SIGN()
           context.system.eventStream.publish(AvailableBalanceChanged(self, d.channelId, d.shortChannelId, commitments1))
-          handleCommandSuccess(c, d.copy(commitments = commitments1)) sending add
+          handleCommandSuccess(c, DATA_NORMAL(d.copy(commitments = commitments1))) sending add
```

Comment:
Let's use lenses? Otherwise it's gonna be tedious when the state class has more than one argument:
```suggestion
          handleCommandSuccess(c, normal.modify(_.data.commitments).setTo(commitments1)) sending add
```

Reply:
Done in https://github.com/ACINQ/eclair/pull/2173/commits/68d479fa9b720c40aa29e93ee669f23b51005007